{{cookiecutter.repo_name}}/Makefile

```diff
@@ -1,121 +1,135 @@
-	@echo "  fake_translations          generate and compile dummy translation files"
-	@echo "  help                       display this help message"
-	@echo "  html_coverage              generate and view HTML coverage report"
-	@echo "  migrate                    apply database migrations"
-	@echo "  dev_requirements           install requirements for local development"
-	@echo "  test_requirements          install requirements for testing"
-	@echo "  quality_requirements       install requirements for quality"
-	@echo "  doc_requirements           install requirements for documentation"
-	@echo "  prod_requirements          install requirements for production"
-	@echo "  pull_translations          pull translations from Transifex"
-	@echo "  push_translations          push source translation files (.po) from Transifex"
-	@echo "  quality                    run Pycodestyle and Pylint"
-	@echo "  pii_check                  check for PII annotations on all Django models"
-	@echo "  requirements               install requirements for local development"
-	@echo "  test                       run tests and generate coverage report"
-	@echo "  validate                   run tests, quality, and PII annotation checks"
-	@echo "  start-devstack             run a local development copy of the server"
-	@echo "  open-devstack              open a shell on the server started by start-devstack"
-	@echo "  pkg-devstack               build the {{cookiecutter.repo_name}} image from the latest configuration and code"
-	@echo "  detect_changed_source_translations       check if translation files are up-to-date"
-	@echo "  validate_translations      install fake translations and check if translation files are up-to-date"
-	@echo ""
-
-clean:
+.PHONY: help clean piptools requirements ci_requirements dev_requirements \
+        validation_requirements doc_requirementsprod_requirements static shell \
+        test coverage isort_check isort style lint quality pii_check validate \
+        migrate html_coverage upgrade extract_translation dummy_translations \
+        compile_translations fake_translations  pull_translations \
+        push_translations start-devstack open-devstack  pkg-devstack \
+        detect_changed_source_translations validate_translations
+
+define BROWSER_PYSCRIPT
+import os, webbrowser, sys
+try:
+	from urllib import pathname2url
+except:
+	from urllib.request import pathname2url
+
+webbrowser.open("file://" + pathname2url(os.path.abspath(sys.argv[1])))
+endef
+export BROWSER_PYSCRIPT
+BROWSER := python -c "$$BROWSER_PYSCRIPT"
+
+# Generates a help message. Borrowed from https://github.com/pydanny/cookiecutter-djangopackage.
+help: ## display this help message
+	@echo "Please use \`make <target>\` where <target> is one of"
+	@perl -nle'print $& if m{^[\.a-zA-Z_-]+:.*?## .*$$}' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}'
```

Comment:
Could you explain what this does?

Reply:
This displays a colorized help message of available `make` targets, kind of like this:

```
Please use `make <target>' where <target> is one of
  clean                     archive and delete most git-ignored files
  detect_changed_source_translations check if translation files are up-to-date
  docs                      build the developer documentation for this repository
  ...
```

Ned came up with a slightly cleaner version that doesn't depend on perl, we should probably switch to that in both cookiecutters: https://github.com/edx/edx-platform/blob/master/Makefile#L9

Reply:
I can't really make out how it's doing that; is it reading this file and getting the targets and descriptions from the comments next to them?

Reply:
Essentially.  Ned wrote a blog post about it (both versions): https://nedbatchelder.com/blog/201804/makefile_help_target.html